api/v1beta1/clusterclass_types.go

```diff
@@ -952,6 +964,7 @@ type PatchSelectorMatchMachineDeploymentClass struct {
 type PatchSelectorMatchMachinePoolClass struct {
 	// names selects templates by class names.
 	// +optional
+	// +kubebuilder:validation:MaxItems=1000
```

Comment:
100 should be ok as well IMO

Reply:
KK. I'll reduce the number of MachineDeployment classes accordingly